### PR TITLE
Feedback for Vidiu

### PR DESCRIPTION
**Pull request by RoseOO**
Needs some more cleanup as it is a little hacky but opening a PR for now because I'm not sure when I'll get around to it again.

Also added some more exception handling to stop it from crashing companion (especially after the Teradek reboots) seems to work well.

Also handles blank passwords. Internally when there is no password set on the Teradek it defaults to admin to authenticate against the API it just never prompts you on the web interface to enter that but the API needs it.

Adds toggle actions for streaming and recording

Ran it for over a week with no issues however, on a Vidiu Pro.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -8,6 +8,10 @@ var request = require('request');
 var cookieJar = request.jar();
 var sessionID = null;
 
+var statusInterval;
```
Comment (krocheck) [inline index.js]: these variables cannot be in global space because they'll be in conflict if people have multiple instances.

Comment (josephdadams) [reply index.js]: @bb21b just making sure you saw this review. :)

Comment (RoseOO) [reply index.js]: @krocheck @josephdadams Thanks for the ping, it'll be a couple of weeks until I get some free time to sort this. But it is good to be busy again! 👍

Comment (krocheck) [reply index.js]: @BB21B let me know when you circle back to this.  I would make the change myself but it would still need to be tested :/

Comment (RoseOO) [reply index.js]: @krocheck Hey I'm going to get back on this. Just getting my brain back in the right gear :P

Comment (josephdadams) [reply index.js]: @bb21b just guessing that my original work on this with the existing variables also need to be moved into the right scope as well. `instance.prototype.statusInterval;` etc.

**Comment by RoseOO** [comment]
Hi all, I think I have this sorted now in 7f47816

**Pull request merged**